src/map/packet_system.cpp

```diff
@@ -4926,6 +4932,21 @@ void SmallPacket0x096(map_session_data_t* const PSession, CCharEntity* const PCh
         return;
     }
 
+    // If the player is already crafting, don't allow them to craft.
+    // This prevents packet injection based multi-craft, or time-based exploits.
+    if (PChar->animation == ANIMATION_SYNTH)
```

Comment:
we might want to do this return if their animation state is anything except "normal".